src/v/cluster/health_monitor_frontend.h

```diff
@@ -71,6 +76,10 @@ class health_monitor_frontend {
     ss::future<cluster_health_overview>
       get_cluster_health_overview(model::timeout_clock::time_point);
 
+    ss::future<std::error_code> maybe_refresh_cluster_health();
+
+    ss::future<> set_disk_health_cb(disk_health_cb_t);
```

Comment:
I think i do not fully understand the need of setting a callback here. Can we just simply query for the state whenever we need it ?

Reply:
Would it be possible to use `disk_space_alert` coming from `node_health_report` ?

Reply:
I did tag you on slack a couple of times for input on this design. Efficiency is important for something called on every produce request. I have some ideas for different ways of doing this, but I'd like to avoid expensive work in the common case. i.e. something about a efficient, run-time, as my solution here.

Reply:
New version incoming that eliminates the callback mechanism.